hydra-cluster/src/CardanoNode.hs

```diff
@@ -125,6 +125,18 @@ getCardanoNodeVersion :: IO String
 getCardanoNodeVersion =
   readProcess "cardano-node" ["--version"] ""
 
+-- | Tries to reach for an existing cardano-node running on default state directory.
+findRunningCardanoNode :: FilePath -> IO (Maybe RunningNode)
+findRunningCardanoNode stateDirectory = do
+  let args'@CardanoNodeArgs{nodeSocket} = defaultCardanoNodeArgs
+  let nodeSocketPath = File nodeSocket
+  maybeSocketPath <- doesFileExist $ unFile nodeSocketPath
```

Comment:
Must: check also if the node is on the right network, fail otherwise.

There _can_ exist `node.socket` files even if the node does not run. Hence, we can use `node.socket` as a first indication, but we **must** try to communicate with the node to deem it present.

Also (and maybe more important) the node running must be serving the right network id. So we querying the tip or some other basic query to verify the expected `NetworkId` is needed here.